### PR TITLE
Restore crystal field parameters after attribute set.

### LINKED ISSUE #21104: Crystal Field: setting peak shape to the multisite function resets field parameters.
### Expected behavior

The multisite function should keep parameter values after the peak shape was set.

### Actual behavior

The parameters are all set to 0.

### Steps to reproduce the behavior

Run this script:
```
from CrystalField import *
cf1 = CrystalField('Pr', 'D3h', Temperature=4, B20=1)
cf2 = CrystalField('Pr', 'C3v', Temperature=4, B20=2)
cf = cf1 + cf2
print cf['ion0.B20']  # 1
print cf['ion1.B20']  # 2
cf.PeakShape = 'Lorentzian'
print cf['ion0.B20']  # 0
print cf['ion1.B20']  # 0
```

### Platforms affected

All

### PR DESCRIPTION
**Pull request by mantid-roman**
Field parameters are saved and restored after setting an attribute that doesn't change the structure of the function (parameter number)

**To test:**

Run script form issue description.

Fixes #21104

**Release Notes** 

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [x] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/CurveFitting/test/Functions/CrystalFieldFunctionTest.h [hunk 1]
```diff
@@ -691,6 +691,19 @@ class CrystalFieldFunctionTest : public CxxTest::TestSuite {
...
+    cf.setAttributeValue("Symmetries", "C2v, D6h");
+    cf.setAttributeValue("Temperatures", std::vector<double>({44, 50}));
+    cf.setAttributeValue("FWHMs", std::vector<double>({1, 2}));
+    cf.setParameter("ion0.B20", 1.0);
+    cf.setParameter("ion1.B20", 2.0);
+    cf.setAttributeValue("PeakShape", "Lorentzian");
+    cf.setParameter("ion0.B20", 1.0);
+    cf.setParameter("ion1.B20", 2.0);
+  }
+
```
Comment (KarlPalmen) [inline Framework/CurveFitting/test/Functions/CrystalFieldFunctionTest.h]: Where are the assertions?

File: Framework/CurveFitting/test/Functions/CrystalFieldFunctionTest.h [hunk 2]
```diff
@@ -699,9 +699,11 @@ class CrystalFieldFunctionTest : public CxxTest::TestSuite {
     cf.setAttributeValue("FWHMs", std::vector<double>({1, 2}));
     cf.setParameter("ion0.B20", 1.0);
     cf.setParameter("ion1.B20", 2.0);
+    TS_ASSERT_EQUALS(cf.getParameter("ion0.B20"), 1.0);
+    TS_ASSERT_EQUALS(cf.getParameter("ion1.B20"), 2.0);
     cf.setAttributeValue("PeakShape", "Lorentzian");
-    cf.setParameter("ion0.B20", 1.0);
-    cf.setParameter("ion1.B20", 2.0);
+    TS_ASSERT_EQUALS(cf.getParameter("ion0.B20"), 1.0);
+    TS_ASSERT_EQUALS(cf.getParameter("ion1.B20"), 2.0);
```
Comment (KarlPalmen) [inline Framework/CurveFitting/test/Functions/CrystalFieldFunctionTest.h]: Great!

**Review by KarlPalmen — APPROVED** [review]
Look good

**Comment by martyngigg** [comment]
It looks like the Python 3 build did finish (checking the console log) but the notification didn't make it back.

**Pull request merged**